packages/material-ui/src/styles/createPalette.js

```diff
@@ -154,32 +154,38 @@ export default function createPalette(palette) {
     return contrastText;
   }
 
-  const augmentColor = (color, mainShade = 500, lightShade = 300, darkShade = 700) => {
+  const augmentColor = ({ color, name, mainShade = 500, lightShade = 300, darkShade = 700 }) => {
     color = { ...color };
     if (!color.main && color[mainShade]) {
       color.main = color[mainShade];
     }
 
-    if (!color.main) {
+    if (!color.hasOwnProperty('main')) {
       throw new MuiError(
-        'Material-UI: The color provided to augmentColor(color) is invalid.\n' +
+        'Material-UI: The color%s provided to augmentColor(color) is invalid.\n' +
```

Comment:
Could you expand the test to include all placeholders? This looks odd to me i.e. I would expect a space between `color` and `%s` and maybe some quotes to identify its origin.